### PR TITLE
[HttpKernel] Merge PartialDenormalizationException errors and validation errors together

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       |
| License       | MIT
| Doc PR        | no needed

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php [hunk 1]
```diff
@@ -52,6 +56,7 @@ final class RequestPayloadValueResolver implements ValueResolverInterface
     public function __construct(
         private readonly SerializerInterface&DenormalizerInterface $serializer,
         private readonly ?ValidatorInterface $validator,
+        private readonly ?TranslatorInterface $translator,
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php]: let's make this default to null and remove the extra args added in tests

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php [hunk 2]
```diff
@@ -71,14 +76,37 @@ public function resolve(Request $request, ArgumentMetadata $argument): iterable
...
+                    $payload = $this->$payloadMapper($request, $type, $attributes[0]);
+                } catch (PartialDenormalizationException $e) {
+                    $trans = $this->translator ? $this->translator->trans(...) : fn ($m, $p) => strtr($m, $p);
+                    foreach ($e->getErrors() as $error) {
+                        $parameters = ['{{ type }}' => implode('|', $error->getExpectedTypes())];
+                        if ($error->canUseMessageForUser()) {
+                            $parameters['hint'] = $error->getMessage();
+                        }
+                        $template = 'This value should be of type {{ type }}.';
+                        $message = $trans($template, $parameters);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php]: missing domain argument: `validators`

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php (hunk 1 above)
Comment (lyrixx) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php]: Same for the validator too ?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php]: Let's do it yes

File: src/Symfony/Component/HttpKernel/composer.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
         "symfony/finder": "^5.4|^6.0",
         "symfony/http-client-contracts": "^2.5|^3",
         "symfony/process": "^5.4|^6.0",
-        "symfony/property-access": "^5.4|^6.0",
+        "symfony/property-access": "^5.4.5|^6.0.5",
```
Comment (lyrixx) [inline src/Symfony/Component/HttpKernel/composer.json:39]: I need this [patch](https://github.com/symfony/symfony/pull/45306)

**Comment by lyrixx** [comment]
@nicolas-grekas thanks for the review. I addressed your comments. I also fixed test for lowest deps

**Review by chalasr — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @lyrixx.

**Pull request merged**